File: server/src/main/java/io/aitchn/dcnucleus/plugin/LoadedPlugin.java
package io.aitchn.dcnucleus.plugin;

import io.aitchn.dcnucleus.api.Plugin;

/**
 * 已載入的插件封裝：
 * - descriptor: 插件描述檔 (plugin.yml)
 * - instance: 插件的主類實例
 * - classLoader: 插件專屬的類別載入器
 */
public class LoadedPlugin {
    private final PluginDescriptor descriptor;
    private final Plugin instance;
    private final PluginClassLoader classLoader;

    public LoadedPlugin(PluginDescriptor descriptor, Plugin instance, PluginClassLoader classLoader) {
        this.descriptor = descriptor;
        this.instance = instance;
        this.classLoader = classLoader;
    }

    public PluginDescriptor getDescriptor() {
        return descriptor;
    }

    public Plugin getInstance() {
        return instance;
    }

    public PluginClassLoader getClassLoader() {
        return classLoader;
    }

    @Override
    public String toString() {
        return "LoadedPlugin{" +
                "descriptor=" + descriptor +
                ", instance=" + instance +
                ", classLoader=" + classLoader +
                '}';
    }
}

File: server/src/main/kotlin/io/aitchn/dcnucleus/server/plugin/LoadedPlugin.kt
package io.aitchn.dcnucleus.server.plugin

import io.aitchn.dcnucleus.api.DCNucleusPlugin
import io.aitchn.dcnucleus.server.plugin.PluginClassLoader
import java.nio.file.Path

data class LoadedPlugin(
    val name: String,
    val jar: Path,
    val classLoader: PluginClassLoader,
    val instance: DCNucleusPlugin
)